tests/ttnn/unit_tests/operations/test_matmul.py

```diff
@@ -7,7 +7,528 @@
+    return best_h, best_w, max_product
+
+
+@pytest.mark.parametrize("n", [1])
+@pytest.mark.parametrize("c", [2])
+@pytest.mark.parametrize("h", [71])
+@pytest.mark.parametrize("w", [35])
+@pytest.mark.parametrize("tile_h", [1, 2, 4, 8, 16, 32])
+@pytest.mark.parametrize("tile_w", [16, 32])
+def test_tiny_tiles(device, n, c, h, w, tile_h, tile_w):
+    torch.manual_seed(0)
+    torch_input_tensor = torch.rand((n, c, h, w), dtype=torch.bfloat16)
+    input_tensor = ttnn.from_torch(
+        torch_input_tensor,
+        tile=(tile_h, tile_w),
+        layout=ttnn.TILE_LAYOUT,
+        device=device,
+        memory_config=ttnn.L1_MEMORY_CONFIG,
+    )
+    output_tensor = ttnn.to_torch(input_tensor)
+    assert_with_pcc(torch_input_tensor, output_tensor, 1)
+
+
+@run_for_wormhole_b0()
+@pytest.mark.parametrize("b", [8])
+@pytest.mark.parametrize("h", [4])
+@pytest.mark.parametrize("m", [256])
+@pytest.mark.parametrize("k", [256])
+@pytest.mark.parametrize("n", [256])
+@pytest.mark.parametrize("tile_h", [16, 32])
+@pytest.mark.parametrize("tile_w", [16, 32])
+@pytest.mark.parametrize("in0_sharded", [True, False])
+@pytest.mark.parametrize("in1_sharded", [True, False])
+@pytest.mark.parametrize("out_sharded", [True, False])
+def test_matmul_reuse_config_sharded_tiny_tile(
+    device, b, h, m, k, n, tile_h, tile_w, in0_sharded, in1_sharded, out_sharded
+):
+    torch.manual_seed(0)
+
+    grid_size = (b, h)
+
+    in0 = torch.ones([b, h, m, k]).bfloat16().float()
+    in1 = torch.randn([b, h, k, n]).bfloat16().float()
+
+    if in0_sharded:
+        in0_memory_config = ttnn.create_sharded_memory_config(
+            (b, h, m, k),
+            core_grid=ttnn.CoreGrid(y=grid_size[1], x=grid_size[0]),
+            strategy=ttnn.ShardStrategy.HEIGHT,
+            orientation=ttnn.ShardOrientation.ROW_MAJOR,
+        )
+    else:
+        in0_memory_config = ttnn.L1_MEMORY_CONFIG
+    in0_t = ttnn.from_torch(
+        in0,
+        tile=(tile_h, 32),
+        dtype=ttnn.bfloat16,
+        layout=ttnn.TILE_LAYOUT,
+        device=device,
+        memory_config=in0_memory_config,
+    )
+
+    if in1_sharded:
+        in1_memory_config = ttnn.create_sharded_memory_config(
+            (b, h, k, n),
+            core_grid=ttnn.CoreGrid(y=grid_size[1], x=grid_size[0]),
+            strategy=ttnn.ShardStrategy.HEIGHT,
+            orientation=ttnn.ShardOrientation.ROW_MAJOR,
+        )
+    else:
+        in1_memory_config = ttnn.L1_MEMORY_CONFIG
+    in1_t = ttnn.from_torch(
+        in1,
+        tile=(32, tile_w),
+        dtype=ttnn.bfloat16,
+        layout=ttnn.TILE_LAYOUT,
+        device=device,
+        memory_config=in1_memory_config,
+    )
+
+    out_block_h = m // tile_h
+    out_block_w = n // tile_w
+    out_subblock_h, out_subblock_w, _ = find_max_subblock(out_block_h, out_block_w)
+
+    program_config = ttnn.MatmulMultiCoreReuseProgramConfig(
+        compute_with_storage_grid_size=grid_size,
+        in0_block_w=k // 32,
```

Comment:
k is parameterized as 256 but your comment says 32 , I guess you could expand the comment.

Reply:
wdym? 
// 32 is the divide